BoxContentSDK/BoxContentSDK/Requests/BOXFileShareRequest.m

```diff
@@ -110,14 +110,24 @@ - (void)performRequestWithCompletion:(BOXFileBlock)completionBlock
     BOXAPIJSONOperation *fileOperation = (BOXAPIJSONOperation *)self.operation;
 
     fileOperation.success = ^(NSURLRequest *request, NSHTTPURLResponse *response, NSDictionary *JSONDictionary) {
+        BOXFile *file = [[BOXFile alloc] initWithJSON:JSONDictionary];
+        if ([self.cacheClient respondsToSelector:@selector(cacheFileShareRequest:withFile:error:)]) {
+            [self.cacheClient cacheFileShareRequest:self
+                                           withFile:file
+                                              error:nil];
+        }
         if (completionBlock) {
-            BOXFile *file = [[BOXFile alloc] initWithJSON:JSONDictionary];
             [BOXDispatchHelper callCompletionBlock:^{
                 completionBlock(file, nil);
             } onMainThread:isMainThread];
         }
     };
     fileOperation.failure = ^(NSURLRequest *request, NSHTTPURLResponse *response, NSError *error, NSDictionary *JSONDictionary) {
+        if ([self.cacheClient respondsToSelector:@selector(cacheFileShareRequest:withFile:error:)]) {
+            [self.cacheClient cacheFileShareRequest:self
+                                           withFile:nil
+                                              error:error];
+        }
```

Comment:
idem. More duplication. I understand that duplication already existed before this change. This change is just following the existing patterns.